code/modules/atmospherics/machinery/airalarm.dm

```diff
@@ -532,13 +532,14 @@
 	if((stat & (NOPOWER|BROKEN)) || shorted)
 		icon_state = "alarmp"
 		return
-	switch(max(danger_level, alarm_area.atmosalm))
-		if (0)
-			src.icon_state = "alarm0"
-		if (1)
-			src.icon_state = "alarm2" //yes, alarm2 is yellow alarm
-		if (2)
-			src.icon_state = "alarm1"
+	if(alarm_area)
```

Comment:
shit, not a fix. I didn't meant to fix that. It's just that I had to make a temporary fix when doing my testing, otherwises this stuff creates so many runtimes that it doesn't show the next potential runtimes.